### ISSUE TITLE
kube-aws s3 bucket with ending "/" breaks upgrade

### ISSUE THREAD

**Issue by jkamenik** [coreos/kube-aws#312]
I ran

```bash
$ kube-aws update --s3-uri s3://foo-bar/
```

After about 15 minutes the cluster was still not available, so I started to do some digging.  `cloud-init` has failed with a 403: Forbidden error

The issue is that the S3 permissions was set to

```json
      {
            "Action": [
                "s3:GetObject"
            ],
            "Resource": "arn:aws:s3:::foo-bar//test-cluster/userdata-controller",
            "Effect": "Allow"
        }
```

The error is caused by the two "//" in the path.

Also, it did not appear as if any new files were uploaded to S3 (but that may just have been because no new files were needed).

----

Solution: I think the ideal solution would either be to "error" if an ending "/" is provided, or simply strip any ending "/" from the s3-uri parameter.

**Comment by mumoshu** [comment]
@jkamenik Good catch, and thanks for the detailed report!
It definitely seems like a bug.
I'm labeling this issue so that we don't forget fixing it.

**Comment by mumoshu** [comment]
This is fixed via #315 which is included since v0.9.3-rc.3.
I'm closing this but please feel free to reopen if the issue still persists!
Thanks for reporting @jkamenik.
